libpromises/exec_tools.c

```diff
@@ -155,7 +155,7 @@ void ActAsDaemon()
                 GetErrorStr());
         }
 
-        if (dup2(fd, STDERR_FILENO) == -1);
+        if (dup2(fd, STDERR_FILENO) == -1)
```

Comment:
@kacf @pasinskim @GregorioDiStefano how can we guard against such stupidity, by using our CI? Is there a simple tool we can run with little effort?